src/structure/pdb.jl

```diff
@@ -30,26 +49,315 @@ end
+    if ba_number==0
+        pdbpath = joinpath(pdb_dir,"$pdbid"*pdbextension[file_format])
     else
-        # Will download error page if ba_number is too high
-        download("http://www.rcsb.org/pdb/files/$pdbid.pdb$ba_number", out_filepath)
+        pdbpath = joinpath(pdb_dir,"$pdbid"*pdbextension[file_format]*"$ba_number")
     end
+    # Download the PDB file only if it does not exist in the "pdb_dir" and when "overwrite" is true
+    if ispath(pdbpath) && !overwrite
+        info("PDB Exists : $pdbid")
+    else
+        info("Downloading PDB : $pdbid")
+        if ba_number == 0            
+            if file_format == PDB || file_format == PDBXML || file_format == mmCIF
+                download("http://files.rcsb.org/download/$pdbid"*pdbextension[file_format]*".gz", archivefilepath)
+            elseif file_format == MMTF
+                # MMTF is downloaded in uncompressed form, thus directly stored in pdbpath
+                download("http://mmtf.rcsb.org/v1.0/full/$pdbid", pdbpath)
+            else
+                warn("Invalid PDB file format!")
+            end
+        else            
+            if file_format == PDB
+                download("http://files.rcsb.org/download/$pdbid"*pdbextension[file_format]*"$ba_number.gz",archivefilepath)
+            elseif file_format == mmCIF
+                download("http://files.rcsb.org/download/$pdbid-assembly$ba_number"*pdbextension[file_format]*".gz", archivefilepath)
+            else
+                warn("Biological Assembly is available only in PDB and mmCIF formats!")
+            end
+        end
+        # Verify if the compressed PDB file is downloaded properly and extract it. For MMTF no extraction is needed
+        if ispath(archivefilepath) && filesize(archivefilepath) > 0 && file_format != MMTF           
+            input = open(archivefilepath) |> ZlibInflateInputStream
+            open(pdbpath,"w") do output
+                for line in eachline(input)
+                    println(output, line)
+                end
+            end
+            close(input)
+        end
+        # Verify if the PDB file is downloaded without any error
+        if !ispath(pdbpath) || filesize(pdbpath)==0
+            throw(ErrorException("Error downloading PDB : $pdbid"))
+        end
+    end
+    rm(archivefilepath)
+end
+
+
+"""
+Downloads a list of PDB files from the RCSB PDB server. 
+By default downloads the PDB files to current working directory; 
+if the keyword argument `pdb_dir` is set the PDB files are downloaded to the
+specified directory;
+if the keyword argument `obsolete` is set `true`, the PDB files are downloaded
+into the obsolete directory inside `pdb_dir`;
+if the keyword argument `overwrite` is set `true`, then it will overwrite the
+PDB file if it exists in the `pdb_dir`.
+"""
+function downloadpdb(pdbidlist::Array{String,1}; pdb_dir::AbstractString=pwd(), file_format::Type=PDB, obsolete::Bool=false, overwrite::Bool=false)
```

Comment:
The last 4 arguments here could just be `kwargs...` so their defaults are defined in one place above.